src/rust/ensogl/lib/components/src/list_view/entry.rs

```diff
@@ -20,266 +19,175 @@ pub const PADDING:f32 = 14.0;
 
-/// The Entry Model Provider for select component.
+// === Entry Provider ===
+
+/// The Entry Provider for [`ListView`] component.
 ///
-/// The select does not display all entries at once, instead it lazily ask for models of entries
-/// when they're about to be displayed. So setting the select content is essentially providing
-/// implementor of this trait.
-pub trait ModelProvider : Debug {
+/// The select does not display all entries at once, instead it lazily ask for entries when they're
+/// about to be displayed. So setting the select content is essentially providing implementor of
+/// this trait.
+pub trait EntryProvider: Debug {
     /// Number of all entries.
     fn entry_count(&self) -> usize;
 
-    /// Get the model of entry with given id. The implementors should return `None` onlt when
-    /// requested id greater or equal to entries count.
-    fn get(&self, id:Id) -> Option<Model>;
+    /// Get the entry with given id. The implementors should return `None` only when the requested
+    /// id is greater or equal to the entry count.
+    fn get(&self, app:&Application, id:Id) -> Option<AnyEntry>;
 }
 
-/// A wrapper for shared instance of some ModelProvider.
+/// A wrapper for shared instance of some EntryProvider.
 #[derive(Clone,CloneRef,Debug,Shrinkwrap)]
-pub struct AnyModelProvider(Rc<dyn ModelProvider>);
+pub struct AnyEntryProvider(Rc<dyn EntryProvider>);
 
-impl<T:ModelProvider + 'static> From<T> for AnyModelProvider {
+impl<T: EntryProvider + 'static> From<T> for AnyEntryProvider {
     fn from(provider:T) -> Self { Self(Rc::new(provider)) }
 }
 
-impl<T:ModelProvider + 'static> From<Rc<T>> for AnyModelProvider {
+impl<T: EntryProvider + 'static> From<Rc<T>> for AnyEntryProvider {
     fn from(provider:Rc<T>) -> Self { Self(provider) }
 }
 
-impl Default for AnyModelProvider {
+impl Default for AnyEntryProvider {
     fn default() -> Self {EmptyProvider.into()}
 }
 
 
-// === Empty Model Provider ===
+// === Empty Entry Provider ===
 
-/// An Entry Model Provider giving no entries.
+/// An Entry Provider giving no entries.
 ///
 /// This is the default provider for new select components.
 #[derive(Clone,CloneRef,Copy,Debug)]
 pub struct EmptyProvider;
 
-impl ModelProvider for EmptyProvider {
-    fn entry_count(&self)          -> usize         { 0    }
-    fn get        (&self, _:usize) -> Option<Model> { None }
-}
-
-
-// === Model Provider for Vectors ===
-
-impl<T:Into<Model> + Debug + Clone> ModelProvider for Vec<T> {
-    fn entry_count(&self) -> usize {
-        self.len()
-    }
+#[derive(Debug,Copy,Clone)]
+enum EmptyProviderEntry {}
 
-    fn get(&self, id:usize) -> Option<Model> {
-       Some(<[T]>::get(self, id)?.clone().into())
+impl display::Object for EmptyProviderEntry {
+    fn display_object(&self) -> &display::object::Instance {
+        match *self {}
     }
 }
 
+impl Entry for EmptyProviderEntry {
+    fn set_focused(&self, _selected: bool) {}
 
-// === Masked Model Provider ===
```

Comment:
The `SingleMaskedProvider` was used in drop down menus to hide the currently selected entry from the list. But that was completely broken. For this reason, I removed the `SingleMaskedProvider`. If we still want that feature then I would add a new issue for it, referring to the old implementation.

Reply:
Was it confirmed with @wdanilo ?

Reply:
Not yet

Reply:
It was used for visualization chooser. The currently selected visualization should not be on the list - and it was not, it was working correctly. So it looks like a regression. Also, Felix, before removing such things, talk about what they are used / are meant to be used for with people. Talking about that on the PR day when there is a deadline of merging that is a bad idea.

Reply:
Adam confirmed that it was not working correctly. But he suggested to include the implementation again and fix it either as part of this PR or later.

@wdanilo Is it okay for you if we include the code, as it was (with some adoptions to the new list views) but do not fix it right now?

Reply:
I would postpone even the restoring old code, as we are really short of time.